object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java

```diff
@@ -464,21 +465,41 @@ private void handleAssignFromRHS(
       if (isVarInDefs(newDefs, (LocalVariableNode) lhs)) {
         ImmutableSet<LocalVarWithTree> setContainingLatestAssignmentPair =
             getSetContainingAssignmentTreeOfVar(newDefs, (LocalVariableNode) lhs);
+        LocalVariableNode localPassedAsMCCParam = getLocalPassedAsMustCallChoiceParam(rhs);
         LocalVarWithTree latestAssignmentPair =
             getAssignmentTreeOfVar(newDefs, (LocalVariableNode) lhs);
-        if (setContainingLatestAssignmentPair.size() == 1) {
-          checkMustCall(
-              setContainingLatestAssignmentPair,
-              typeFactory.getStoreBefore(node),
-              "variable overwritten by assignment " + node.getTree());
+        if (localPassedAsMCCParam != null
+            && setContainingLatestAssignmentPair.stream()
+                .map(localVarWithTree -> localVarWithTree.localVar.getElement())
+                .anyMatch(elem -> elem.equals(localPassedAsMCCParam.getElement()))) {
+          // If the rhs is MCC with the lhs, replace latestAssignmentPair with this Assignment
```

Comment:
I'm not sure I understand what is going on in this case.  Can you write an example here showing why exactly we need this new case?  It seems like code below around line 512 should handle adding the assignment to the set in this case.  Maybe we need to fix a bug down there?

Reply:
There was a bug around line 512! Instead of replacing the previous assignment node, we were just adding the new assignment node.
https://github.com/kelloggm/object-construction-checker/pull/282/commits/4cd278c17fdf8a8d5032bd4b16aeea842d8e3d69